tsdb/db.go

```diff
@@ -760,52 +770,82 @@ func (db *DB) Compact() (err error) {
 		// consistently, we explicitly remove the last value
 		// from the block interval here.
 		head := NewRangeHead(db.head, mint, maxt-1)
-		if err := db.compactHead(head); err != nil {
+		t, err := db.compactHead(head)
+		if err != nil {
 			return err
 		}
+		if t > checkpointTime {
+			checkpointTime = t
+		}
+	}
+
+	checkpointAttempted = true
```

Comment:
If head was not compactable at first `for` iteration, is it correct we try to perform checkpoint?

Reply:
Right, we should not :) thanks for pointing out